components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FxaDeviceConstellation.kt

```diff
@@ -138,9 +138,13 @@ class FxaDeviceConstellation(
 
     override fun setDeviceNameAsync(name: String): Deferred<Boolean> {
         return scope.async {
-            handleFxaExceptions(logger, "changing device name") {
+            val rename = handleFxaExceptions(logger, "changing device name") {
                 account.setDeviceDisplayName(name)
             }
+            // See the latest device (name) changes after changing it.
+            val refreshDevices = deviceManager.refreshDevicesAsync().await()
+
+            rename && refreshDevices
```

Comment:
this may be an overkill: a failing `refreshDevices` doesn't mean that we failed to update the name. But, let's roll with this.

Reply:
It's also interesting to consider what you do when device names misalign. `name` was 'foo`, but after refreshing our device record from the server, we saw `bar`. Is that a failure? Maybe not necessarily, but it's worth logging at least!

So yeah - let's check if our names align, and log an `info` success and a `warn` on `name!=newName` (probably without logging actual device names).

Reply:
> It's also interesting to consider what you do when device names misalign. `name` was 'foo`, but after refreshing our device record from the server, we saw `bar`. Is that a failure?

Our observer should be called so we update the UI to show what we get from the API.

> So yeah - let's check if our names align, and log an `info` success and a `warn` on `name!=newName` (probably without logging actual device names).

Good call for logging this case. It _shouldn't_ happen though.

Reply:
Ah, the logging would have to be done in Fenix since that's where we observe the device constellation changes.